presto-spi/src/main/java/com/facebook/presto/spi/block/ByteArrayBlock.java

```diff
@@ -158,6 +158,11 @@ public Block copyRegion(int positionOffset, int length)
     {
         checkValidRegion(getPositionCount(), positionOffset, length);
 
+        if (positionOffset == 0 && length == positionOffset &&
+                arrayOffset == 0 && positionCount == values.length) {
+            return this;
```

Comment:
Instead of changing how `Block::copyRegion()` behaves (and changing all `Block` implementations), can't we just introduce a `isCompact()` method in `Block` like you did with `Slice` and let the caller determine whether to copy or not?